core/services/gateway/handlers/common/ratelimiter.go

```diff
@@ -7,19 +7,27 @@ import (
 	"golang.org/x/time/rate"
 )
 
+const (
+	defaultWorkflowRPS   = 5.0
+	defaultWorkflowBurst = 50
+)
+
 // Wrapper around Go's rate.Limiter that supports both global and a per-sender rate limiting.
 type RateLimiter struct {
-	global    *rate.Limiter
-	perSender map[string]*rate.Limiter
-	config    RateLimiterConfig
-	mu        sync.Mutex
+	global      *rate.Limiter
+	perSender   map[string]*rate.Limiter
+	perWorkflow map[string]*rate.Limiter
```

Comment:
This is a big no no IMO -- we can't mix domains like this

Once we are in the gateway domain proper (as opposed to in the connector handler which is a capabilities-specific handler for the gateway) then we concepts like workflow etc shouldn't creep in at all